frontend/components/bc/BcInputSearch.vue

```diff
@@ -0,0 +1,322 @@
+  if (currentIndex.value >= resultItems.value.length) {
+    currentIndex.value = 0
+    return
+  }
+  if (currentIndex.value < 0) {
+    currentIndex.value = resultItems.value.length - 1
+    return
+  }
+}
+
+const {
+  bottom,
+  left,
+  right,
+  update: updatePositionOfInput,
+} = useElementBounding(elementInput, {
+  immediate: false,
+})
+
+const showPopover = () => {
+  updatePositionOfInput()
+  hasPopover.value = true
+  if (!elementResults.value || elementResults.value.matches(':popover-open')) return
+  if (props.shouldSelectFirstResult) currentIndex.value = 0
+  elementResults.value.showPopover()
+}
+const hidePopover = () => {
+  hasPopover.value = false
+  currentIndex.value = -1
+  if (!elementResults.value || !elementResults.value.matches(':popover-open')) return
+  elementResults.value.hidePopover()
+}
+
+const isEmpty = computed(() => !props.results?.length)
+watchDebounced(input, async () => {
+  emit('search', input.value ?? '')
+})
+watch(hasInput, () => {
+  if (!hasInput.value) {
+    hidePopover()
+  }
+})
+const hasErrorOrNoResults = computed(() => props.hasError || props.results === undefined)
+watch([
+  () => props.results,
+  () => props.hasError,
+], () => {
+  if (!hasInput.value) return
+  showPopover()
+})
+watchEffect(() => {
+  elementResults.value?.style.setProperty('--position-top', bottom.value + 'px')
+  elementResults.value?.style.setProperty('--position-left', left.value + 'px')
+  elementResults.value?.style.setProperty('--position-right', right.value + 'px')
+})
+
+const handleEsc = (event: KeyboardEvent) => {
+  if (!hasInput.value) return
+  event.stopPropagation()
+  if (!hasPopover.value) return
+  event.preventDefault()
+  hidePopover()
+}
+const handleFocus = (event: Event) => {
+  if (!(event.target as HTMLInputElement).value) return
+  if (hasResults.value) showPopover()
+}
+
+const selectedResult = computed(() => props.results?.[currentIndex.value])
+const handleSubmit = () => {
+  if (!selectedResult.value) return
+  emit('submit', toRaw(selectedResult.value))
+  if (props.shouldClearOnSubmit) input.value = ''
+  hidePopover()
+}
+const handleClick = (index: number) => {
+  currentIndex.value = index
+  elementInput.value?.focus()
+  handleSubmit()
+}
+const isResultsHovered = () => elementResults.value?.matches(':hover')
+const handleBlur = () => {
+  if (isResultsHovered()) {
+    return
+  }
+  hidePopover()
+}
+onMounted(() => {
+  if (props.hasFocus) elementInput.value?.focus()
+})
+const ariaActivedescendant = computed(() => {
+  if (props.hasError) return `${idResults}-error`
+  if (isEmpty.value) return `${idResults}-empty`
+  return `${idElementList}-${currentIndex.value}`
+})
+</script>
+
+<template>
+  <form
+    role="search"
+    class="bc-input-search__form"
+    @keydown.arrow-up.prevent="hasResults && changeSelectedIndex(-1)"
+    @keydown.arrow-down.prevent="hasResults && changeSelectedIndex(+1)"
+    @keydown.arrow-down.alt.exact.prevent="hasResults && showPopover()"
```

Comment:
Is this really necessary?  
```suggestion
    @keydown.arrow-up="hasResults && changeSelectedIndex(-1)"
    @keydown.arrow-down="hasResults && changeSelectedIndex(+1)"
    @keydown.arrow-down.alt.exact="hasResults && showPopover()"
```